dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/AbstractImageRegistry.java

```diff
@@ -158,37 +159,70 @@
         // Get all the tools based on the found namespaces
         List<Tool> apiTools = getToolsFromNamespace(namespaces);
 
+        String registryString = getRegistry().getDockerPath();
+
         // Add manual tools to list of api tools
         User user = userDAO.findById(userId);
-        List<Tool> manualTools = toolDAO.findByMode(ToolMode.MANUAL_IMAGE_PATH);
+        List<Tool> manualTools = toolDAO.findByModeRegistryNamespace(ToolMode.MANUAL_IMAGE_PATH, registryString, organization);
+        List<Tool> notManualTools = toolDAO.findByNotModeRegistryNamespace(ToolMode.MANUAL_IMAGE_PATH, registryString, organization);
+        apiTools.addAll(manualTools);
+
+        // Update api tools with build information
+        updateAPIToolsWithBuildInformation(apiTools);
+
+        // Update db tools by copying over from api tools
+        List<Tool> newDBTools = updateTools(apiTools, notManualTools, user, toolDAO);
+
+        // Get tags and update for each tool
+        for (Tool tool : newDBTools) {
+            logToolRefresh(dashboardPrefix, tool);
 
-        // Get all tools in the db for the given registry
-        List<Tool> dbTools = new ArrayList<>(getToolsFromUser(userId, userDAO, toolDAO));
+            List<Tag> toolTags = getTags(tool);
+            final SourceCodeRepoInterface sourceCodeRepo = SourceCodeRepoFactory
+                .createSourceCodeRepo(tool.getGitUrl(), client, bitbucketToken == null ? null : bitbucketToken.getContent(),
+                    gitlabToken == null ? null : gitlabToken.getContent(), githubToken.getContent());
```

Comment:
Can't githubToken be null? If someone logs in with Google? Not that we have any users like that, but maybe someday...